### PR TITLE
[CAPI] Add attention, mol attention and multi head attention to CAPI @open sesame 08/30 14:47

### PR DESCRIPTION
**Pull request by songgot**
[CAPI] Add attention, mol attention and multi head attention to CAPI
- Add LAYER_ATTENTION
- Add LAYER_MOL_ATTENTION
- Add LAYER_MULTI_HEAD_ATTENTION

**Self evaluation:**
1. Build test:   [X]Passed [ ]Failed [ ]Skipped
2. Run test:     [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: hyunil park <hyunil46.park@samsung.com>

### CODE REVIEW

File: api/nntrainer-api-common.h [hunk 1]
```diff
@@ -53,6 +53,10 @@ typedef enum {
   ML_TRAIN_LAYER_TYPE_RNNCELL = 22,  /**< RNN Cell Layer type (Since 7.0) */
   ML_TRAIN_LAYER_TYPE_ZONEOUTLSTMCELL =
     23, /**< ZoneoutLSTM Cell Layer type (Since 7.0) */
+  ML_TRAIN_LAYER_TYPE_ATTENTION = 24,     /**< Attention Layer type */
```
Comment (lhs8928) [inline api/nntrainer-api-common.h:56]: The enumerations of attention type is already added in #1990.
How about rebase onto this pr?

Comment (songgot) [reply api/nntrainer-api-common.h:56]: okay thank you.

**Review by DonghakPark — APPROVED** [review]
LGTM

**Review by jihochu — APPROVED** [review]
LGTM

**Comment by lhs8928** [comment]
This code already included in #1960

**Pull request closed without merging**